lib/queue.rb

```diff
@@ -1,31 +1,45 @@
 class Queue
 
-  def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+  def initialize(size = 22)
+    @size = size
+    @store = Array.new(size)
+    @head = @back = 0
   end
 
   def enqueue(element)
```

Comment:
👍 , I know it's silly, but you probably want to prevent someone from pushing `nil` into the queue because you're using the value of `nil` in `@head` to indicate if the Queue is empty or not.